pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyService.java

```diff
@@ -414,5 +428,16 @@ public Authentication getProxyClientAuthenticationPlugin() {
         return this.proxyClientAuthentication;
     }
 
+    public void addPrometheusRawMetricsProvider(PrometheusRawMetricsProvider metricsProvider) {
+        if (metricsServlet == null) {
+            if (pendingMetricsProviders == null) {
```

Comment:
if we want to support concurrency here, then we must ensure consistency here.

so:
- use `synchronised` to  guard accesses to `metricsServlet` and to `pendingMetricsProviders`
- use a threadsafe List implementation and not `LinkedList` (this could not be needed if we guard with `synchronised` every access to the list

Reply:
Yes the whole state machine must be synchronised. And so it's not necessary to use a concurrent structure.